types/set/README.md

```diff
@@ -0,0 +1,137 @@
+# Set
+
+_Module_
+
+Implements a very simple `SetInt` struct.
```

Comment:
Could we add a little section explaining its use cases and purpose? I'll do this in the other types too.